source/load-balancers/load-balancers.rst

```diff
@@ -0,0 +1,553 @@
+
+If a :code:`getMore` fails with a network error, drivers MUST leave the connection pinned
+to the cursor. When the cursor's :code:`close` method is invoked, drivers MUST NOT execute
+a :code:`killCursors` command because the pinned connection is no longer valid and MUST
+return the connection back to the pool.
+
+Connection Pinning
+^^^^^^^^^^^^^^^^^^
+
+For drivers that implement a connection pool and when using a pinned connection, the driver MUST
+emit only 1 :code:`ConnectionCheckOutStartedEvent`, and only 1 :code:`ConnectionCheckedOutEvent`
+or :code:`ConnectionCheckOutFailedEvent`. Similarly, the driver MUST only publish 1
+:code:`ConnectionCheckedInEvent`.
+
+When executing a transaction in load balancing mode, drivers MUST follow the rules outlined
+in `Sharded Transactions <../transactions/transactions.rst#sharded-transactions>`__ with one
+exception: drivers MUST use the same connection for all commands in the transaction
+(excluding retries of commitTranscation and abortTransaction in some cases). Pinning
+to a single connection ensures that all commands in the transaction target the same
+service behind the load balancer. The rules for pinning to a connection and releasing
+a pinned connection are the same as those for server pinning in non-load balanced sharded
+transactions as described in `When to unpin <../transactions/transactions.rst#when-to-unpin>`__.
+
+Connection Tracking
+^^^^^^^^^^^^^^^^^^^
+
+The driver connection pool MUST track the purpose for which connections are checked out
+in the following 3 categories:
+
+- Connections checked out for cursors
+- Connections checked out for transactions
+- Connections checked out for operations not falling under the previous 2 categories
+
+When the connection pool's :code:`maxPoolSize` is reached and the pool times out waiting
+for a new connection the :code:`WaitQueueTimeoutError` MUST include a new detailed message,
+"Timeout waiting for connection from the connection pool. maxPoolSize: n, connections
+in use by cursors: n, connections in use by transactions: n, connections in use by other
+operations: n". A :code:`ConnectionCheckOutFailedEvent` MUST also be published with the
+same message.
+
+
+Error Handling
+--------------
+
+Initial Handshake Errors
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+If there is a network error or timeout on the connection before the hello or legacy hello completes,
+the driver MUST NOT make any changes to the :code:`TopologyDescription` or mark the server as Unknown.
+The driver MUST also close the associated connection. If the network error occurred on the handshake's
+:code:`hello` command, the connection pool MUST NOT be cleared. If the network error occurred after
+the handshake's hello command, the connection POOL must clear close all connections with the matching
+:code:`serviceId`.
+
+Post-Handshake Errors
+^^^^^^^^^^^^^^^^^^^^^^
+
+When the driver is in load balanced mode and encounters any error described here as a state
```

Comment:
The "described here" sentence doesn't link to anything. Maybe we can avoid linking at all by saying "When the driver is operating in load balanced mode and an application operation receives a state change error, the driver MUST NOT...".

Reply:
Updated per suggestion.